backend/app/routes/admin/changePassword.js

```diff
@@ -0,0 +1,63 @@
+const to = require('await-to-js').default;
+const argon2 = require('argon2');
+const Admin = require('../../models/Admin');
+const { ErrorHandler } = require('../../../helpers/error');
+const constants = require('../../../constants');
+
+module.exports = async (req, res, next) => {
+  const { oldPassword, newPassword } = req.body;
+
+  const userRecord = await Admin.findOne({
+    email: res.locals.decode.email,
+  });
+
+  const isPasswordCorrect = await argon2.verify(
+    userRecord.passwordHash,
+    oldPassword
+  );
+
+  if (!isPasswordCorrect) {
+    const error = new ErrorHandler(constants.ERRORS.INPUT, {
+      statusCode: 400,
+      message: 'Wrong old password',
+      user: userRecord.email,
+      errStack: 'Password change',
+    });
+    return next(error);
+  }
+
+  if (oldPassword === newPassword) {
+    const error = new ErrorHandler(constants.ERRORS.INPUT, {
+      statusCode: 400,
+      message: 'Old password cannot be same as new password.',
+      user: userRecord.email,
+      errStack: 'Password change',
+    });
+    return next(error);
+  }
+
+  const hashedPassword = await argon2.hash(newPassword);
+
+  const [err, user] = await to(
+    Admin.findOne({ email: userRecord.email }, () => {
```

Comment:
you should not use findOne and then save method. You know which fields to update, so you should use find and update method. It saves time eventually resulting in better latency